packages/studioCMSBlog/src/pages/blog/[slug].astro

```diff
@@ -0,0 +1,46 @@
+---
+import { BlogLayout } from "studiocms-blog:layouts";
+import { PostHeader } from "studiocms-blog:components";
+import { StudioCMSRenderer, contentHelper } from "studiocms:components";
+
+// Get the slug from the URL
+const { slug } = Astro.params;
+
+const fourOhFour = `${import.meta.env.BASE_URL}404`
+
+// If no slug is provided, redirect to 404
+if (!slug) {
+    return Astro.redirect(fourOhFour);
```

Comment:
I think it may be better practice to return a `404` instead: `return new Response(null, { status: 404 })`